### ISSUE TITLE
"track files like this in Git LFS" flow when committing

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#6055]
**Please describe the problem you think should be solved**

Following on from #997 - which is our way of helping the user not have their pushes blocked by GitHub because of the file size limit - there's the potential to help users get their Git LFS configured better.

In the classic apps, we'd show them the list of rules in "Repository Settings" - which is tracked as #69 - but being more pro-active about this to catch them _before_ they need to rewrite history is something I'd like to explore.

**[Optional] Do you have any potential solutions in mind?**

No strong ideas currently, but here's some thought bubbles:

 - we know GitHub will [warn on pushes](https://help.github.com/articles/working-with-large-files/#conditions-for-large-files) for files larger than 50MB - we could use that threshold to help guide people to add rules to Git LFS
 - we can use a similar flow to #6013 and suggest rules to add
 - catching users before they commit is much easier than having to fix up a rejected push

**Comment by HenrikoMagnifico** [comment]
Any updates regarding LFS in Github Desktop?

**Comment by TerryChan** [comment]
Any updates on LFS in GitHub Desktop?
Do you have any time plan on implementing this feature?

**Comment by shiftkey** [comment]
@HenrikoMagnifico @TerryChan apologies for not noticing this earlier.

Not sure I understand the context. What Git LFS-related features do you believe are not working within Desktop currently?

**Comment by billygriffin** [comment]
@shiftkey What are your thoughts on combining this and #69? They feel sufficiently similar in what problem is being solved, just with #69 having the solution from the previous apps.

**Comment by shiftkey** [comment]
@billygriffin #69 came from the old way of managing Git LFS files, whereas this idea feels like it could be its own flow, similar to or building upon the work done to address #997.

Essentially I still see this issue as an opportunity for us to guide/help users as part of their normal commit workflows, rather than just letting them edit their settings. Because of that they feel like they could be done at separate times, so merging them together into one issue doesn't make much sense to me.

**Comment by TerryChan** [comment]
> @HenrikoMagnifico @TerryChan apologies for not noticing this earlier.
> 
> Not sure I understand the context. What Git LFS-related features do you believe are not working within Desktop currently?

@shiftkey Thanks for your reply. I mean the feature that configure the files that are tracked by Git LFS directly within GitHub Desktop. Since I found that I can't configure tracked LFS files in GitHub Desktop directly, and it's a missing feature from the classic apps.
As steveward answered [here](https://github.com/desktop/desktop/issues/6373#issuecomment-445836634), looks like you have the plan to complete this feature. So do you have any time plan on **when** to implement this feature? Or do you have any thoughts and ideas on **how** to implement this feature?

**Comment by freshyill** [comment]
> @HenrikoMagnifico @TerryChan apologies for not noticing this earlier.
> 
> Not sure I understand the context. What Git LFS-related features do you believe are not working within Desktop currently?

Even just a warning that LFS doesn't work would be an improvement. Right now it just spins, literally forever, with no feedback. It looks like it's actually doing something, but it's apparently not. 

I found this out the hard way. I read lots of warnings that pushing to LFS was slow. Netlify's docs even say "very, very slow". Activity monitor showed network traffic from git-lfs, so I assumed everything was working, but this first push would take a while. After 16 hours or so, I had a connection hiccup, and it failed.

The next time, I pushed from the command line. The command line noted that it was pushing to LFS, and yes, it was _a little_ slower than a regular push. A warning that this wasn't supported would have saved 12+ hours.

**Comment by billygriffin** [comment]
@freshyill Can you help me understand better what wasn't working and what specifically you did? Git LFS should work in GitHub Desktop today - this issue is about configuring specific files in the app to use LFS as opposed to having to manage that outside the context of Desktop, so I'm curious to understand whether you weren't able to push code that was *already* configured in LFS, or what was causing the issue.

**Comment by jwvanderbeck** [comment]
Not sure if this is already done as part of the existing GitLFS support (I'm still getting up to speed using the new client) but one thing that would be great was if the choice of gitignore settings could influence the initial setup.  For example if I choose a Unity preset when creating a new repo, it would be reasonable to assume that I will have large files like models, textures, audio, video, etc that could intelligently be pre-setup.

**Comment by billygriffin** [comment]
Because this is the same underlying issue as #69, I'm going to close that one. One possibility initially _might be_ to just allow users to manually configure files in preferences. However, we might be able to be smarter about this as this issue alludes to and as @shiftkey described [above](https://github.com/desktop/desktop/issues/6055#issuecomment-463595211), and incorporate a better flow here into the commit workflow when committing large files.

**Comment by AITYunivers** [comment]
This is never coming is it?

**Comment by Flavelius** [comment]
It would be an improvement already to atleast have a window to configure tracked files like sourcetree does. This could have presets or a suggestion popup to be pre-populated with sane choices on first open per repo. But this is optional, the more important part is having an easier way to input tracked files in the first place, other than editing the config manually. The sourcetree-way is quite usable even for non-git-knowledgeable people

**Comment by CyberPunkRules** [comment]
If you Only knew….

C. Salcido
________________________________

**Comment by CyberPunkRules** [comment]
My accounts been compromised.

Christopher Salcido

________________________________

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/18298.
